source/fundamentals/transactions.txt

```diff
@@ -0,0 +1,130 @@
+read or write operations that you intend to run sequentially. Sessions
+enable :manual:`causal consistency
+</core/read-isolation-consistency-recency/#causal-consistency>` for a
+group of operations or allow you to execute operations in an
+:website:`ACID transaction </basics/acid-transactions>`. MongoDB
+guarantees that the data involved in your transaction operations remains
+consistent, even if the operations encounter unexpected errors.
+
+When using the {+driver-short+}, you can create a new session from a 
+``Client`` instance as a ``ClientSession``. We recommend that you reuse
+your client for multiple sessions and transactions instead of
+instantiating a new client each time.
+
+.. warning::
+
+   Use a ``Session`` only with the ``Client`` (or associated
+   ``Database`` or ``Collection``) that created it. Using a
+   ``Session`` with a different ``Client`` results in operation
+   errors.
+
+Methods
+-------
+
+Create a ``ClientSession`` by using the ``startSession()`` method on your
+``Client`` instance. You can then modify the session state by using the
+following methods:
+
+.. list-table::
+   :widths: 25 75
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``startTransaction()``
+     - | Starts a new transaction for this session with the
+         default transaction options. You cannot start a
+         transaction if there's already an active transaction
+         on the session.
+       |
+       | To set transaction options, use ``startTransaction(transactionOptions: TransactionOptions)``.
+
+   * - ``abortTransaction()``
+     - | Ends the active transaction for this session. Returns an error 
+         if there is no active transaction for the
+         session or the transaction was previously ended.
+
+   * - ``commitTransaction()``
+     - | Commits the active transaction for this session. Returns an
+         error if there is no active transaction for the session or if the
+         transaction was ended.
+
+A ``Session`` also has methods to retrieve session properties and modify
+mutable session properties. View the `API documentation <{+api+}/apidocs/mongodb-driver-kotlin-coroutine/mongodb-driver-kotlin-coroutine/com.mongodb.kotlin.client.coroutine/-client-session/index.html>`__
+to learn more about these methods.
+
+Example
+-------
+
+The following example demonstrates how you can create a session, create a transaction,
+and commit a changes to existing documents:
+
+1. Create a session from the client using the ``startSession()`` method.
+#. Use the ``startTransaction()`` method to start a transaction.
+#. Update the specified documents, then use the ``commitTransaction()`` method if all
+   operations succeed, or ``abortTransaction()`` if any operations fail.
+
+This example uses the following {+language+} data classes to model its documents:
+
+.. literalinclude:: /examples/generated/TransactionsTest.snippet.data-class.kt
+   :language: kotlin
+
```

Comment:
Apologies for multiple rounds of feedback here – now that I see this on the page I feel that we need some sort of buffer between this code block and the example. I think I'd lift this to the top of the section, before "The following example demonstrates how..."